pkg/cmd/codespace/ssh.go

```diff
@@ -526,27 +571,36 @@ func (a *App) printOpenSSHConfig(ctx context.Context, opts sshOptions) (err erro
 			result := sshResult{}
 			defer wg.Done()
 
-			session, err := codespaces.ConnectToLiveshare(ctx, a, noopLogger(), a.apiClient, cs)
+			codespaceConnection, err := codespaces.GetCodespaceConnection(ctx, a, a.apiClient, cs)
 			if err != nil {
 				result.err = fmt.Errorf("error connecting to codespace: %w", err)
-			} else {
-				defer safeClose(session, &err)
-
-				invoker, err := rpc.CreateInvoker(ctx, session)
-				if err != nil {
-					result.err = fmt.Errorf("error connecting to codespace: %w", err)
-				} else {
-					defer safeClose(invoker, &err)
-
-					_, result.user, err = invoker.StartSSHServer(ctx)
-					if err != nil {
-						result.err = fmt.Errorf("error getting ssh server details: %w", err)
-					} else {
-						result.codespace = cs
-					}
-				}
+				sshUsers <- result
+				return
+			}
+
+			fwd, err := portforwarder.NewPortForwarder(ctx, codespaceConnection)
+			if err != nil {
+				result.err = fmt.Errorf("failed to create port forwarder: %w", err)
+				sshUsers <- result
+				return
 			}
 
+			invoker, err := rpc.CreateInvoker(ctx, fwd)
```

Comment:
We've got the pattern of "create a PortForwarder and then create an Invoker from it" a lot now - which I know is an artifact of the migration - but we might be able to move the PF creation into `rpc.CreateInvoker` now and save some repetition. 

Just thinking out loud though, can take it or leave it, or do it later if you prefer

Reply:
All of the commands that call `CreateInvoker` end up using the port forwarder further down the line, plus it makes mocking the port forwarder much easier for the invoker tests so I figured it would be better to just pass it in rather than create a new one 🙂